polyfill/lib/datetime.mjs

```diff
@@ -444,19 +441,9 @@ export class DateTime {
     };
     const roundingIncrement = ES.ToTemporalRoundingIncrement(options, maximumIncrements[smallestUnit], false);
 
-    let { deltaDays, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.DifferenceTime(
-      GetSlot(this, HOUR),
-      GetSlot(this, MINUTE),
-      GetSlot(this, SECOND),
-      GetSlot(this, MILLISECOND),
-      GetSlot(this, MICROSECOND),
-      GetSlot(this, NANOSECOND),
-      GetSlot(other, HOUR),
-      GetSlot(other, MINUTE),
-      GetSlot(other, SECOND),
-      GetSlot(other, MILLISECOND),
-      GetSlot(other, MICROSECOND),
-      GetSlot(other, NANOSECOND)
+    let { days: deltaDays, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = calendar.timeUntil(
+      this.toTime(),
```

Comment:
These should be abstract operations so that the call to `toTime` isn't observable (though feel free to defer this to your next PR)

Reply:
Will do!